compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -1250,7 +1250,7 @@ pub fn install_ice_hook(bug_report_url: &'static str, extra_info: fn(&Handler))
         #[cfg(windows)]
         if let Some(msg) = info.payload().downcast_ref::<String>() {
             if msg.starts_with("failed printing to stdout: ") && msg.ends_with("(os error 232)") {
-                early_error_no_abort(ErrorOutputType::default(), &msg);
+                early_error_no_abort(ErrorOutputType::default(), msg.as_str());
```

Comment:
Good catch, that one's not necessary.

Reply:
Sorry, I was misled by this code being within a `#[cfg(windows)]`, and I'm compiling on Linux. It's the same story as the other two below: `DiagnosticMessage` doesn't impl `From<&String>`. I removed that impl in #110579 because it was resulting in unnecessary allocations at lots of call sites.